### ISSUE TITLE
MSBuild static property functions

### ISSUE THREAD

**Issue by gerboengels** [Caphyon/clang-power-tools#246]
We're using vcpkg to generate a NuGet package. This generated package has a .targets file which uses `$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), .vcpkg-root))` inside a variable definition. Unfortunately, this construct is not recognized by your build script.

Is it possible to add support for these kind of functions (or this function in particular)?

See also https://msdn.microsoft.com/en-us/library/dd633440.aspx#BKMK_GetDirectoryNameOfFileAbove for documentation and an example

The actual snippet where it is used, is as follows:
```xml
  <PropertyGroup>
    <VcpkgRoot Condition="'$(VcpkgRoot)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), .vcpkg-root))\installed\x64-windows\</VcpkgRoot>
  </PropertyGroup>
  <ItemDefinitionGroup>
    <ClCompile>
      <AdditionalIncludeDirectories>%(AdditionalIncludeDirectories);$(VcpkgRoot)include</AdditionalIncludeDirectories>
    </ClCompile>
  </ItemDefinitionGroup>
```

**Comment by cpp-red-lion** [comment]
Hi,

Thanks for bringing this issue to our attention 👍
We'll address it soon and update this thread when it's done.

**Comment by sdwe** [comment]
Hi, we have the same issue. Our concrete usage pattern is:
```
  <PropertyGroup Label="Globals">
    <ProductRootFolder>$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), ProductRootFolder.props))\</ProductRootFolder>
  </PropertyGroup>
```

**Comment by hero101111** [comment]
Hello,

I'm happy to inform you that we've just added support for `[MSBuild]::GetDirectoryNameOfFileAbove`.
It will be included in the next release. Thanks for the feedback.

Regards,
Gabriel

---

Please note that this fix is currently live only for the github repository.
The latest Visual Studio extension (`3.0.1`) does not contain the fix. It will be included in the next release.

Until then, as a workaround, you can manually replace `clang-build.ps1` in the VS extension directory with the latest from github. To locate the extension directory search for `clang-build.ps1` in `%LOCALAPPDATA%`

Don't forget to restart Visual Studio after replacing the script file.

**Comment by cpp-red-lion** [comment]
This was fixed in `v3.1` released today.
